src/functions/edh-marshaller.ts

```diff
@@ -3,72 +3,119 @@ import {
   DynamoDBStreamEvent,
   Handler,
 } from 'aws-lambda';
-import { AWSError, SQS } from 'aws-sdk';
-import { SQSService } from '../services/sqs';
-import { PromiseResult } from 'aws-sdk/lib/request';
-import { SendMessageResult } from 'aws-sdk/clients/sqs';
-import { debugOnlyLog, getTargetQueueFromSourceARN, getTechRecordEnvVar } from '../utils/utilities';
+import { SQS } from 'aws-sdk';
+import AWSXray from 'aws-xray-sdk';
+import { SendMessageRequest } from 'aws-sdk/clients/sqs';
+import { debugOnlyLog, getTechRecordEnvVar } from '../utils/utilities';
 import { transformTechRecord } from '../utils/transform-tech-record';
+import { BatchItemFailuresResponse } from '../models/BatchItemFailures';
+
+let sqs: SQS;
 
 /**
  * λ function to process a DynamoDB stream and forward it to an appropriate SQS queue.
  * @param event - DynamoDB Stream event
  * @param context - λ Context
  * @param callback - callback function
  */
-const edhMarshaller: Handler = async (event: DynamoDBStreamEvent): Promise<void | Array<PromiseResult<SendMessageResult, AWSError>>> => {
-  const processFlatTechRecords: boolean = getTechRecordEnvVar();
+const edhMarshaller: Handler = async (event: DynamoDBStreamEvent): Promise<BatchItemFailuresResponse> => {
+  const res: BatchItemFailuresResponse = {
+    batchItemFailures: [],
+  };
 
   if (!event) {
     console.error('ERROR: event is not defined.');
-    return undefined;
+    return res;
   }
+
   const records: DynamoDBRecord[] = event.Records;
+
   if (!records || !records.length) {
     console.error('ERROR: No Records in event: ', event);
-    return undefined;
+    return res;
   }
 
   debugOnlyLog('Records: ', records);
 
-  // Instantiate the Simple Queue Service
-  const sqsService: SQSService = new SQSService(new SQS());
-  const sendMessagePromises: Promise<PromiseResult<SendMessageResult, AWSError>>[] = [];
+  const sendMessagePromises = [];
 
+  if (!sqs) {
+    sqs = AWSXray.captureAWSClient(
+      new SQS({
+        region: 'eu-west-1',
+        apiVersion: '2012-11-05',
+      }),
+    );
+  }
```

Comment:
Maybe this is a bit simpler? Saves you the `if` nesting.
```suggestion
  sqs = sql ?? AWSXray.captureAWSClient(
    new SQS({
      region: 'eu-west-1',
      apiVersion: '2012-11-05',
    }),
  );
```

Reply:
In terms of readability I think I prefer the if check as with the null coalesce it feels weird doing self assignment if 
`sqs` already defined.